### ISSUE TITLE
Add “delete branch” support

### ISSUE THREAD

**Issue by BasThomas** [rnystrom/GitHawk#1542]
![GitHawk Upload by BasThomas](https://i.imgur.com/U6RAqgj.jpg)

Just had this (https://github.com/twof/CLSwift/issues/9) merged but couldn’t delete the branch from GitHawk. Would be nice maybe? Or too out-of-scope?

<details>
<summary>Bug Report Dump (Auto-generated)</summary>
<pre>
Version 1.18.0 (1518753170)
Device: iPhone X (iOS 11.2.5)
TestFlight: true
</pre>
</details>

**Comment by rnystrom** [comment]
Nah I’d love this. Does the API exist?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BasThomas** [comment]
Oh, that's a good question.

**Comment by BasThomas** [comment]
You would probably have to do some logic to check if a pull request is merged and all, but yes: https://developer.github.com/v3/git/refs/#delete-a-reference

**Comment by rnystrom** [comment]
Awesome! Merged state already exists, makes this super easy! We could literally add a floating button. Tho need to make sure the merged branch is part of the same repo.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
Could also be nice to have an Auto Delete Branch setting to handle this automatically after a merge. Don't know if you guys are open to unique features 😛

**Comment by rnystrom** [comment]
@SD10 hell ya we are! That’s a really cool idea. Anything that speeds up repo management is fair game. I want to get GitHawk into a state where I’d rather not use the .com 😇

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BasThomas** [comment]
> Tho need to make sure the merged branch is part of the same repo.

That's including forks right? 😃

Should be noted that if something else still depends on said branch, we shouldn't show the button. Is there an easy way to check that?

**Comment by rnystrom** [comment]
@BasThomas I’m not sure. For instance, I can’t delete branches from PRs made from forks. That’s what you mean right?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
@rnystrom For example, you can't delete a branch that has an open PR against it. There are definitely some things to consider. I'm not sure how the API would handle such a situation.

**Comment by BasThomas** [comment]
Yeah, but the other way around. In this case (see image), I forked a repo which was merged; then couldn't delete the branch (on my fork).

**Comment by genebean** [comment]
Just wanted to add a big +1 for this.
